python/ray/data/tests/test_dataset_binary.py

```diff
@@ -66,10 +66,10 @@ def test_read_binary_files_with_fs(ray_start_regular_shared):
 def test_read_binary_files_with_paths(ray_start_regular_shared):
     with gen_bin_files(10) as (_, paths):
         ds = ray.data.read_binary_files(paths, include_paths=True, parallelism=10)
-        for i, (path, item) in enumerate(ds.iter_rows()):
-            assert path == paths[i]
+        for i, row in enumerate(ds.iter_rows()):
```

Comment:
after making single-column unwrap as same as before, when the schema has multiple columns (path and binary), the way to access each column still has to be changed as here. So it breaks backwards compatiblity. It's due to the underlying difference between simple vs Arrow block when iterating rows.

I am thinking an alternative fix is to add a parameter into `read_binary_files(output_in_python_list_format=False)`, by default it outputs in Arrow block. User can opt-in to output simple block.

WDYT? @ericl and @clarkzinzow.

Reply:
I'm ok with this change as it is a beta API, though we should warn about it in release notes at least.